components/sso-hostobject/org.wso2.carbon.hostobjects.sso/src/main/java/org/wso2/carbon/hostobjects/sso/internal/builder/LogoutRequestBuilder.java

```diff
@@ -100,4 +153,113 @@ public LogoutRequest buildLogoutRequest(String subject, String reason,
 
         return logoutReq;
     }
+
+    /**
+     * Overload Logout request for sessionIndexId is not exist case
+     * @param subject
+     * @param reason
+     * @param issuerId
+     * @return
+     */
+    public LogoutRequest buildSignedLogoutRequest(String subject, String reason,
+            String issuerId, int tenantId, String tenantDomain, String destination) throws Exception {
+        Util.doBootstrap();
+        LogoutRequest logoutReq = new org.opensaml.saml2.core.impl.LogoutRequestBuilder().buildObject();
+        logoutReq.setID(Util.createID());
+
+        DateTime issueInstant = new DateTime();
+        logoutReq.setIssueInstant(issueInstant);
+        logoutReq.setNotOnOrAfter(new DateTime(issueInstant.getMillis() + 5 * 60 * 1000));
+
+        IssuerBuilder issuerBuilder = new IssuerBuilder();
+        Issuer issuer = issuerBuilder.buildObject();
+        issuer.setValue(issuerId);
+        logoutReq.setIssuer(issuer);
+
+        NameID nameId = new NameIDBuilder().buildObject();
+        nameId.setFormat(SSOConstants.SAML2_NAME_ID_POLICY);
+        nameId.setValue(subject);
+        logoutReq.setNameID(nameId);
+
+        logoutReq.setReason(reason);
+        logoutReq.setDestination(destination);
+
+        SSOAgentCarbonX509Credential ssoAgentCarbonX509Credential =
+                new SSOAgentCarbonX509Credential(tenantId, tenantDomain);
+        setSignature(logoutReq, SignatureConstants.ALGO_ID_SIGNATURE_RSA,
+                new X509CredentialImpl(ssoAgentCarbonX509Credential));
+
+        return logoutReq;
+    }
+
+    /**
+     * Sign the SAML LogoutRequest message
+     *
+     * @param logoutRequest
+     * @param signatureAlgorithm
+     * @param cred
+     * @return
+     */
+
+
```

Comment:
Please remove above blank lines as we don't need spaces between method and its comment.